src/packageurl/__init__.py

```diff
@@ -346,27 +346,25 @@ class PackageURL(
     https://github.com/package-url/purl-spec
     """
 
-    name: str
-    namespace: Optional[str]
-    qualifiers: Union[str, Dict[str, str], None]
-    subpath: Optional[str]
     type: str
+    namespace: Optional[str]
+    name: str
     version: Optional[str]
+    qualifiers: Dict[str, str]
+    subpath: Optional[str]
 
     def __new__(
-        self,
+        cls,
         type: Optional[AnyStr] = None,
         namespace: Optional[AnyStr] = None,
-        name: Optional[AnyStr] = None,  # NOQA
+        name: Optional[AnyStr] = None,
         version: Optional[AnyStr] = None,
-        qualifiers: Union[AnyStr, Dict[str, str], None] = None,
+        qualifiers: Optional[Union[AnyStr, Dict[str, str]]] = None,
         subpath: Optional[AnyStr] = None,
-    ) -> "PackageURL":  # this should be 'Self' https://github.com/python/mypy/pull/13133
-        required = dict(type=type, name=name)
-        for key, value in required.items():
-            if value:
-                continue
-            raise ValueError(f"Invalid purl: {key} is a required argument.")
+    ) -> Self:
+        for arg in type, name:
+            if not arg:
+                raise ValueError(f"Invalid purl: {arg} is a required argument.")
```

Comment:
Reminder: revert

Reply:
I initially rewrote it as:

```python
for arg in "type", "name":
    if not locals().get(arg):
        raise ValueError(f"Invalid purl: {arg} is a required argument.")
```

but ultimately decided to just set it back to how it was originally